certificate-service/app/src/main/java/se/inera/intyg/certificateservice/application/unit/service/GetUnitMessagesService.java

```diff
@@ -0,0 +1,89 @@
+import se.inera.intyg.certificateservice.application.unit.service.validator.GetUnitMessagesRequestValidator;
+import se.inera.intyg.certificateservice.domain.certificate.model.Certificate;
+import se.inera.intyg.certificateservice.domain.message.model.Message;
+import se.inera.intyg.certificateservice.domain.unit.service.GetUnitMessagesDomainService;
+
+@Service
+@RequiredArgsConstructor
+public class GetUnitMessagesService {
+
+  private final GetUnitMessagesRequestValidator getUnitMessagesRequestValidator;
+  private final ActionEvaluationFactory actionEvaluationFactory;
+  private final MessagesRequestFactory messagesRequestFactory;
+  private final GetUnitMessagesDomainService getUnitMessagesDomainService;
+  private final CertificateConverter certificateConverter;
+  private final QuestionConverter questionConverter;
+  private final ResourceLinkConverter resourceLinkConverter;
+
+  public GetUnitMessagesResponse get(GetUnitMessagesRequest request) {
+    getUnitMessagesRequestValidator.validate(request);
+
+    final var actionEvaluation = actionEvaluationFactory.create(
+        request.getUser(),
+        request.getUnit(),
+        request.getCareUnit(),
+        request.getCareProvider()
+    );
+
+    final var messagesRequest = messagesRequestFactory.create(
+        request.getMessagesQueryCriteria()
+    );
+
+    final var messagesResponse = getUnitMessagesDomainService.get(messagesRequest,
+        actionEvaluation);
+
+    return GetUnitMessagesResponse.builder()
+        .questions(
+            messagesResponse.messages().stream()
+                .map(message ->
+                    questionConverter.convert(
+                        message,
+                        message.actions(actionEvaluation, getCertificateForMessage(
+                                message,
+                                messagesResponse.certificates()
+                            )
+                        )
+                    )
+                )
+                .toList()
+        )
+        .certificates(messagesResponse.certificates().stream()
+            .map(certificate -> certificateConverter.convert(
+                    certificate,
+                    certificate.actionsInclude(Optional.of(actionEvaluation)).stream()
+                        .map(certificateAction ->
+                            resourceLinkConverter.convert(
+                                certificateAction,
+                                Optional.of(certificate),
+                                actionEvaluation
+                            )
+                        )
+                        .toList()
+                )
+            )
+            .toList()
+        )
+        .build();
+  }
+
+  private Certificate getCertificateForMessage(Message message, List<Certificate> certificateList) {
+    return certificateList.stream()
+        .filter(certificate -> certificate.id().id().equals(message.certificateId().id()))
+        .findAny()
```

Comment:
I was interessted in knowing the difference between findAny and findFirst. From google:

```Java
The findAny() method returns any element from a Stream, while
the findFirst() method returns the first element in a Stream

This is to allow for maximal performance in parallel operations.
```

In this case, this should not matter right? So I think its fine use findAny!